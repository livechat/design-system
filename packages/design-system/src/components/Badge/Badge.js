import * as React from 'react';
import * as PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './style.scss';

function Badge({ children, className: extraClassName, secondary, ...props }) {
  const className = cx(styles.badge, {
    [styles['badge--secondary']]: secondary,
    ...(extraClassName ? {[extraClassName]: true} : {})
  });

  return (
    <span className={className} {...props}>
      {children}
    </span>
  );
}

Badge.propTypes = {
  children: PropTypes.node.isRequired,
  secondary: PropTypes.bool
};

export default Badge;
