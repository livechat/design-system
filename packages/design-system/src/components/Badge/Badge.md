### Badge

## Usage

Use badges to draw attention errors and new, updated or important content, like unread messages. Hide badhe when the count is 0.

```js
<p>
  <Badge>7</Badge> Tickets
</p>
<p>
  <Badge>!</Badge> Settings
</p>
<p>
  <Badge>99+</Badge> Chats
</p>
```

## Badge placement

Place badges directly adjoining the related UI object. Badges can appear to the right or left of an UI object

```js
<Badge>!</Badge>
```

## Over 99

If a badge needs to display a number above 99, use “99+”

## Style

Badges are small enough to fit next to an UI object. They have a bold, and filled style that makes them stand out, and a fully rounded border radius.

## Color

- Red reserved to display a sense of urgency or indicate state.
- Light Variation is best for non-essential element on the screen and should recede from the primary content.

# Primary

Errors and important content, like unread messages, new tickets.

```js
<Badge>1</Badge>
<Badge>11</Badge>
<Badge>99+</Badge>
```

Non-essential content like new features or updates.

```js
<Badge secondary>1</Badge>
<Badge secondary>11</Badge>
<Badge secondary>99+</Badge>
```
