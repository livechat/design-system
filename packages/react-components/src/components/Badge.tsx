import * as React from "react";
import cx from "classnames";

export interface IBadgeProps {
  children: React.ReactElement;
  secondary: boolean;
  className?: string;
}

export const Badge = ({
  children,
  className: extraClassName = "",
  secondary = false,
}: IBadgeProps) => {
  const className = cx("lc-badge", {
    "lc-badge--secondary": secondary,
    [extraClassName]: !!extraClassName,
  });

  return <span className={className}>{children}</span>;
};
