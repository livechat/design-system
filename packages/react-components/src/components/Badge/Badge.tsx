import * as React from 'react';
import cx from 'classnames';

export interface IBadgeProps {
  secondary?: boolean;
  className?: string;
}

export const Badge: React.FC<IBadgeProps> = ({
  children,
  className: extraClassName = '',
  secondary = false,
}) => {
  const className = cx('lc-badge', {
    'lc-badge--secondary': secondary,
    [extraClassName]: !!extraClassName,
  });

  return <span className={className}>{children}</span>;
};
