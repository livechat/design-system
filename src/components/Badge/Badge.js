import * as React from 'react';
import * as PropTypes from 'prop-types';
import styles from './style.scss';

function Badge(props) {
  const badgeType = `badge--${props.type || 'primary'}`;

  return (
    <span className={`${styles.badge} ${styles[badgeType]}`}>
      {props.children}
    </span>
  );
}

Badge.propTypes = {
  children: PropTypes.node.isRequired,
  /**
   * Type of badge
   */
  type: PropTypes.oneOf(['primary', 'light'])
};

export default Badge;
