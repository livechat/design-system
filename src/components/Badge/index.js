export { default as Button } from './Badge';
