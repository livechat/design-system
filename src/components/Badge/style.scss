@import '../../scss/variables/colors';

.badge {
  position: relative;
  display: inline-block;
  vertical-align: middle;
  height: 16px;
  min-width: 16px;
  box-sizing: border-box;
  font-size: 10px;
  line-height: 1.5;
  border-radius: 16px;
  overflow: hidden;
  text-align: center;
  padding-left: 3px;
  padding-right: 3px;

  & + & {
    margin-left: 4px;
  }
}

.badge--primary {
  color: $badge-primary-color;
  background-color: $badge-primary-bg;
}

.badge--light {
  color: $badge-light-color;
  background-color: $badge-light-bg;
}
