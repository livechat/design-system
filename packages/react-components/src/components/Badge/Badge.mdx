import { Meta, Title, Subtitle, Canvas, ArgTypes } from '@storybook/blocks';
import { Figma } from '@storybook/addon-designs/blocks';
import * as BadgeStories from './Badge.stories';
import {
  BlockColumnsContainer,
  BlockColumns,
} from '../../stories/components/BlockColumns';

<Meta of={BadgeStories} />

<Title>Badge</Title>

[Component API](#ComponentAPI) | [Content Spec](#ContentSpec)

<Canvas of={BadgeStories.Default} />

#### Example implementation

```jsx
<Badge size="medium" count={1} />
```

## Component API <a id="ComponentAPI" />

<ArgTypes />

## Content Spec <a id="ContentSpec" />

<Figma
  showLink
  height="100%"
  url="https://www.figma.com/file/9FDwjR8lYvincseDkKypC4/Component-Documentations?type=design&node-id=731-59622&mode=dev"
/>
