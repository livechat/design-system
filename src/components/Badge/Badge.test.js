import * as React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

describe('Button', () => {
  it('should render primary badge', () => {
    const component = shallow(<Badge>!</Badge>);

    expect(component).toMatchSnapshot();
  });

  it('should render light badge', () => {
    const component = shallow(<Badge type="light">!</Badge>);

    expect(component).toMatchSnapshot();
  });
});
