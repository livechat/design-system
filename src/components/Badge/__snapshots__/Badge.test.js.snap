// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render light badge 1`] = `
<span
  className="badge badge--light"
>
  !
</span>
`;

exports[`Button should render primary badge 1`] = `
<span
  className="badge badge--primary"
>
  !
</span>
`;
